python/pysrc/kaskada/sources/arrow.py

```diff
@@ -81,6 +82,7 @@ def __init__(
             rows: One or more rows represented as dicts.
             time_column: The name of the column containing the time.
             key_column: The name of the column containing the key.
+            retained: Whether added rows should be retained for queries.
```

Comment:
We could (possibly) document the interaction with `rows` / materializations.

Reply:
This might benefit from more explanation, and some help on when to use it vs when not to.